docs/2.0/docs/patcher/tutorials/applying-first-patch.md

```diff
@@ -1,3 +1,163 @@
+* Pushing that unit up to your remote repository in GitHub
+* Triggering patcher to detect the out of date unit and apply the patch
+* Reviewing the pull request created by Patcher and merging the changes.
+
+### The `patcher-test` Module
+
+In this tutorial we'll use the `patcher-test` module from the `gruntwork-io/terraform-aws-utilities` repository. This module has a version `v0.10.3` that we'll intentionally specify as outdated in our infrastructure unit, and have patcher update to v0.10.4 which includes a demonstration breaking change.
+
+## Applying a patch using Patcher locally
+
+Choose a directory named: <CustomizableValue id="DIRECTORY" /> in your repository where we'll add a new infrastructure unit.  In that folder create a file called `terragrunt.hcl` with the following content:
+
+```hcl title="$$DIRECTORY$$/terragrunt.hcl"
+
+terraform {
+  source = "git@github.com:gruntwork-io/terraform-aws-utilities.git//modules/patcher-test?ref=v0.10.3"
+}
+
+# Include the root `terragrunt.hcl` configuration, which has settings common across all environments & components.
+include "root" {
+  path = find_in_parent_folders()
+}
+```
+
+Now we'll run `patcher update`.  This can be run from the root of the repository, where it will scan every unit in all sub-folders, or directly inside the <CustomizableValue id="DIRECTORY" /> directory.
+
+```bash
+
+$ cd $$DIRECTORY$$
+$ patcher update
+```
+
+
+![Patcher Update homepage showing an update to the patcher-test module is available](/img/patcher/tutorials/patcher_update_preview.png)
+
+From here you can hit enter to update `patcher-test`, or `u` to view usages of the module, `v` to view the changelog, or `q` to quit.
+
+<details>
+<summary>Hitting `u` to view changes shows the following:</summary>
+
+![The changes page shows every unit that uses the module, and what the most recent version it](/img/patcher/tutorials/patcher_update_usages.png)
+</details>
+
+<details>
+<summary>Hitting `v` to view the changelog shows the following:</summary>
+
+![The changelog page shows the changelog directly from the upstream module](/img/patcher/tutorials/patcher_update_changelog.png)
+</details>
+
+From here hit `enter` and patcher will apply the update:
+
+![Patcher update shows that it found a patch to be applied and what version it is incrementing to](/img/patcher/tutorials/patcher_update_in_progress.png)
+
+![Patcher Update completion notice](/img/patcher/tutorials/patcher_update_complete.png)
+
+<!-- spell-checker: disable -->
+You can now inspect the filesystem and see the results of the patch.  In this case it resulted in changing the version of `patcher-test` from `v0.10.3` to `v0.10.5` and added a new required argument `sampleinput` to the unit. This change can now be committed and pushed to your repository.
+<!-- spell-checker: enable -->
```

Comment:
out of curiosity. what was it that spell-checker didn't like?

Reply:
`sampleinput` isn't a valid word apparently